File: Largest_Number_At_Least_Twice_of_Others.cpp
class Solution {
public:
    int dominantIndex(vector<int>& nums) {
        int maxi =-1; // finding the max element;;
        for(int i=0;i<nums.size(); i++)
        {
            maxi= max(maxi, nums[i]);
        }
        int ans =-1;
        for(int i=0; i<nums.size(); i++)
        {
            if(maxi != nums[i]  &&   nums[i] * 2 > maxi) return -1;
            if(maxi == nums[i])
            {
                ans = i;
            }
        }
        return ans;

    }
};

File: Largest_Number_At_Least_Twice_of_Others.java
class Solution {
    public int dominantIndex(int[] nums) {
            int max=0,index=-1;
            for(int i=0;i<nums.length;i++){
                if(nums[i]>max){
                    max=nums[i];
                    index=i;
                    }
                    }
                    for(int i=0;i<nums.length;i++){
                        if(i!=index && nums[i]>(max/2) ){
                            return -1;
                            }
                            }
            return index;
    }
}
